tests/NgTests/Validators/PackageHasSignatureValidatorFacts.cs

```diff
@@ -0,0 +1,317 @@
+using Moq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using NgTests.Infrastructure;
+using NuGet.Configuration;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Services.Metadata.Catalog;
+using NuGet.Services.Metadata.Catalog.Helpers;
+using NuGet.Services.Metadata.Catalog.Monitoring;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test.Catalog;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test.Exceptions;
+using NuGet.Versioning;
+using Xunit;
+
+namespace NgTests
+{
+    public class PackageHasSignatureValidatorFacts
+    {
+        public class ShouldRunValidatorAsync : FactsBase
+        {
+            [Fact]
+            public async Task SkipsIfAllEntriesAreDeletes()
+            {
+                var context = CreateValidationContext(
+                    catalogEntries: new[]
+                    {
+                        new CatalogIndexEntry(
+                            uri: null,
+                            type: DeleteCatalogEntry,
+                            commitId: string.Empty,
+                            commitTs: DateTime.MinValue,
+                            id: PackageId,
+                            version: PackageNuGetVersion),
+                        new CatalogIndexEntry(
+                            uri: null,
+                            type: DeleteCatalogEntry,
+                            commitId: string.Empty,
+                            commitTs: DateTime.MinValue,
+                            id: PackageId,
+                            version: PackageNuGetVersion),
+                    });
+
+                Assert.False(await _target.ShouldRunValidatorAsync(context));
+            }
+
+            [Fact]
+            public async Task RunsIfNotAllEntriesAreDeletes()
+            {
+                var context = CreateValidationContext(
+                    catalogEntries: new[]
+                    {
+                        new CatalogIndexEntry(
+                            uri: null,
+                            type: DeleteCatalogEntry,
+                            commitId: string.Empty,
+                            commitTs: DateTime.MinValue,
+                            id: PackageId,
+                            version: PackageNuGetVersion),
+                        new CatalogIndexEntry(
+                            uri: null,
+                            type: DetailsCatalogEntry,
+                            commitId: string.Empty,
+                            commitTs: DateTime.MinValue,
+                            id: PackageId,
+                            version: PackageNuGetVersion),
+                    });
+
+                Assert.True(await _target.ShouldRunValidatorAsync(context));
+            }
+        }
+
+        public class RunValidatorAsync : FactsBase
+        {
+            [Fact]
+            public async Task ReturnsGracefullyIfLeafsHaveSignatureFile()
```

Comment:
Leaves?

Reply:
Good catch. I should go back to school 😄